plugins/extract/pipeline.py

```diff
@@ -92,35 +96,50 @@ def load_aligner(aligner, loglevel, configfile, normalize_method):
+        logger.debug("Loading Masker: '%s'", masker_name)
+        masker = PluginLoader.get_masker(masker_name)(loglevel=loglevel,
+                                                      configfile=configfile)
+        return masker
+
     def set_parallel_processing(self, multiprocess):
-        """ Set whether to run detect and align together or separately """
+        """ Set whether to run detect, align, and mask together or separately """
         detector_vram = self.detector.vram
         aligner_vram = self.aligner.vram
-
-        if detector_vram == 0 or aligner_vram == 0:
-            logger.debug("At least one of aligner or detector have no VRAM requirement. "
-                         "Enabling parallel processing.")
-            return True
+        masker_vram = self.masker.vram
 
         if not multiprocess:
             logger.debug("Parallel processing disabled by cli.")
             return False
 
+        if detector_vram == 0 or aligner_vram == 0 or masker_vram == 0:
+            logger.debug("At least one of aligner, detector or masker have no VRAM requirements. "
+                         "Enabling parallel processing.")
+            return True
+
+
+
         gpu_stats = GPUStats()
         if gpu_stats.is_plaidml and (not self.detector.supports_plaidml or
-                                     not self.aligner.supports_plaidml):
-            logger.debug("At least one of aligner or detector does not support plaidML. "
+                                     not self.aligner.supports_plaidml or
+                                     not self.masker.supports_plaidml):
+            logger.debug("At least one of aligner, detector or masker does not support plaidML. "
                          "Enabling parallel processing.")
             return True
 
         if not gpu_stats.is_plaidml and (
-                (self.detector.supports_plaidml and aligner_vram != 0) or
-                (self.aligner.supports_plaidml and detector_vram != 0)):
-            logger.warning("Keras + non-Keras aligner/detector combination does not support "
-                           "parallel processing. Switching to serial.")
+               (self.detector.supports_plaidml and aligner_vram != 0) or
+               (self.aligner.supports_plaidml and detector_vram != 0) or
+               (self.masker.supports_plaidml and masker_vram != 0)):
+            logger.warning("Keras + non-Keras aligner/detector/masker combination does not "
+                           "support parallel processing. Switching to serial.")
             return False
 
-        if self.detector.supports_plaidml and self.aligner.supports_plaidml:
+        if self.detector.supports_plaidml and (self.aligner.supports_plaidml and
+                                               self.masker.supports_plaidml):
             logger.debug("Both aligner and detector support plaidML. Disabling parallel "
                          "processing.")
```

Comment:
This is not quite correct. The idea here is to switch to serial processing if more as one plugin/worker does support plaidml because we can't reliable get the available vram.
I'd just remove this part and change the first check (line 126) to something like:
```
 if gpu_stats.is_plaidml:
    return bool(self.detector.supports_plaidml) + bool(self.aligner.supports_plaidml) + bool(self.masker.supports_plaidml)) <= 1
```